src/rgw/rgw_trim_mdlog.cc

```diff
@@ -677,9 +677,49 @@ class MetaPeerTrimPollCR : public MetaTrimPollCR {
   {}
 };
 
+namespace {
+bool sanity_check_endpoints(const DoutPrefixProvider *dpp, rgw::sal::RadosStore* store) {
+  bool retval = true;
+  auto current = store->svc()->mdlog->get_period_history()->get_current();
+  const auto& period = current.get_period();
+  for (const auto& [_, zonegroup] : period.get_map().zonegroups) {
+    if (zonegroup.endpoints.empty()) {
```

Comment:
i don't think mdlog trimming ever tries to use zonegroup endpoints. it's fine to log a warning anyway, but this case probably shouldn't be fatal

Reply:
Fixed! Thank you.